### ISSUE TITLE
quizz_score is not calculated correctly

### ISSUE THREAD

**Issue by tarteo** [odoo/odoo#43508]
It does not take "Multiple choice: only one answer" into account.
https://github.com/odoo/odoo/blob/d3489146e9e1e20903bf8775527cb8b32a2ba383/addons/survey/models/survey_user.py#L70-L73

Example:
My colleague got a score of 82.1% for the Odoo Certification 13 but this is not possible if you can have a maximum score of 80 points (1 right answer is worth 1 point and there are 80 "multiple choice: only one answer" questions). Also one wrong answer is worth -0.5 points
This means there is one answer worth 2 points or one question with 2 answers both worth 1 point (2 points in total).

To prove 82.1% (rounded):
(71-4.5)/81=82.1
(71-4.5)/80=83.1% (if the maximum score is correct; 80 points)

So my colleagues actual score on the Odoo Certification should be 83.1% 
It is possible some people didn't get their certification due to this bug. To be exact if you got a score of 56 points (16 errors, and 64 rights answers) currently you'll get a score of 69.1% (so you don't get the certification) but it should be 70% (in this case you would get the certification).

To be clear imagine your whole exam consist of one "Multiple choice: only one answer" question. With the following choices and scores:

**What is the capital city of the Netherlands:**
Amsterdam score: 1 (the actual correct answer)
Rotterdam score: 1
Breda score: -0.5

This means by the calculation (mentioned earlier) the `total_possible_score` is 2. But if you answer this question you can only select one answer this means the highest possible score would be 50% (1 point).

**Comment by pedrobaeza** [comment]
Thanks for the analysis. Is the exam using actual code of this repository or is it something private? If so, I think you should contact by mail with the problem to the exam responsibles (not sure if there's any contact address). If not, try to reproduce the problem in runbot.

**Comment by Yenthe666** [comment]
I actually believe they reuse survey for the exams now. @jubodoo is responsible for e-learning so he can verify this.

**Comment by yung-wang** [comment]
tested in Runbot [13.0]

A certification with 1 question.
![image](https://user-images.githubusercontent.com/30628967/72712124-bc4daa00-3b6a-11ea-98bf-1d12eb6cf087.png)

A question with type **Multiple choice: only one answer**. And 2 answer marked as 'is a correct answer', this is somehow allowed in Odoo.
![image](https://user-images.githubusercontent.com/30628967/72712240-ea32ee80-3b6a-11ea-9066-39929a546da5.png)

If you run the survey, you can only score 50%.
![image](https://user-images.githubusercontent.com/30628967/72712346-249c8b80-3b6b-11ea-9850-295860b59fd2.png)

score:
![image](https://user-images.githubusercontent.com/30628967/72712373-30884d80-3b6b-11ea-9ae4-aa4759576b43.png)

So Odoo allows you to select 2 answer as 'is a correct answer', but you're only able to select 1 with the type **Multiple choice: only one answer** during the test.
Seems more like a bug than a configuration fault.

**Comment by dbeguin** [comment]
Indeed, we don't constrain to only one correct answer in `Multiple choice: only one answer` question type. Scoring is based on computing the maximum points we can get for the question, and making the ratio between selected answer(s) score on max score. If the user configures wrongly the answers (scoring + is correct), then it obviously leads to inconsistent situations. (Especially in the last example where Rotterdam is not the capital of Netherlands.. with The Hague instead, this could have been discussed :)  
So no, this is not a bug, it's kind of known behaviour. For the Odoo 13 certification, indeed, there might be an error in the configuration. To check with @jubodoo.

To avoid this kind of issue in the future.. it's not that easy. We have an in between situation here.
- If we add the constraint, we will limit the capabilities of the survey : a question like
```
Select one town in Netherlands : 
* Amsterdam (is correct)
* The Hague (is correct)
* Paris
* New-York
```
will not be possible anymore. I guess there are some other use cases where this kind of 'only one correct answer is enough to pass the question' can be used.

- Or we make an exception for this question type in the scoring computation, which will lead to a lot of complexity.
What if scoring is different between Amsterdam (2 points) and the Hague (1 points) (because Amsterdam is the real capital, and the Hague is kind of the administrative capital..) ?
    * Should it be taken into account or totally ignored ? 
    * If one correct answer is selected, then score = 2+1 ? 
    * Or = score of the selected answer ? In that case, we should remove the 'not-selected correct answer' from the total survey max score..
Do we want to handle this, no.

So to avoid incoherent configuration, we could add the constraint.
@jubodoo : what do you think ?

**Comment by yung-wang** [comment]
I think we make it way to complex to think like that.

A survey with 1 question with type **Multiple choice: only one answer** 
```
Select one town in Netherlands : 
* Amsterdam (is correct) (2 points)
* The Hague (is correct) (1 points)
* Paris
* New-York
```

So maximum score is 3 points. **But you are only allowed to select 1 answer.**

If you chose Amsterdam your score will be:
2/3 = 66%

If you chose The Hague your score will be:
1/3 = 33%

So according your test, you will never be able to achieve 100%.

For this kind scoring you need to use **Multiple choice: multiple answers**

**Comment by yung-wang** [comment]
Something to think about and some use cases.

```
Select one town in Netherlands : 
* Amsterdam (is correct) (2 points)
* The Hague (is correct) (1 points)
* Paris
* New-York
```

If you really want to allow one answer, and Amsterdam or The Hagues is correct, but Amsterdam is rated higher. Than I think this would be the best scoring.
For Amsterdam. 2/2 = 100%
For The Hague. 1/1 = 100%
Both answer are correct. No matter which one you select.

If you want a different use case 
```
Select a town in Netherlands. To score extra, select the town closest to Rotterdam: 
* Amsterdam (is correct) (1 points)
* The Hague (is correct) (2 points)
* Paris
* New-York
```
If you only allow one answer, and Amsterdam or The Hagues is correct, but The Hague is rated higher. Than I think this would be the best scoring.
For Amsterdam. 1/2 = 50%
For The Hague. 2/2 = 100%
Because max is 2 points.

**Comment by Yenthe666** [comment]
Any update @dbeguin?

**Comment by jethroguce** [comment]
up this, also having weird behavior for this, and could also not understand the logic for this.
